discovery_engine_core/web-api/src/models.rs

```diff
@@ -112,7 +112,24 @@ impl DocumentId {
     }
 }
 
-#[derive(Clone, Debug, Display, Serialize, Deserialize, PartialEq, Eq, Hash, AsRef)]
+impl<'de> Deserialize<'de> for DocumentId {
```

Comment:
❓ this looks quite elegant, but if i'm not mistaken, this is not yet called anywhere in the routes? currently the parameter extraction reads the id as a `String`,  then some url decoding and calls the `new()` constructor, but deserialization into `DocumentId` is not called.

the deserialization is called when we get data from the database though, but it seems unnecessary to check validity again if we assume that we only store valid data in the first place.

it seems that we either have to change the routing to extract the id parameter as a `DocumentId` instead of a `String` (not sure if that's doable) or we need to add the `is_id_valid()` check to the `DocumentId::new()` (and remove it from the serde as this is to a certain extent in vain).

Reply:
It is called with the ingestion endpoint and any other endpoint taking ids in a json blob.

In actix parsing from path uses `FromStr`.

The postgres database uses `Encode` and sqlx transparent, so it shouldn't affect this.

But when I proposed this I had forgotten that we also get document ids from elastic instead of just giving them to elastic :face_exhaling: 

Maybe we need two `DocumentId` types or something like this. 

I which we could have context dependent serialization etc. similar to how you can use context objects in scala for some things.

Reply:
> In actix parsing from path uses FromStr.

To avoid confusion with `ToString` we probably should not include the `urldecode` part in `FromStr` but provide some `Urldecode` wrapper which does so, like `Path<(Urldecode<DocumentId>, Urldecode<DocumentPropertyId>)>`. Maybe there is already a helper type like that idk.